cli/azd/pkg/project/service_target_dotnet_containerapp.go

```diff
@@ -393,6 +396,26 @@ func (at *dotnetContainerAppTarget) Deploy(
 		if err != nil {
 			return nil, fmt.Errorf("deploying bicep template: %w", err)
 		}
+		// set the name of the container app to the name of the deployment
+		if len(deploymentResult.Resources) == 0 {
+			return nil, fmt.Errorf("no resources found after deployment. Expected at least one Azure Container App")
+		}
+		acaCount := 0
+		for _, res := range deploymentResult.Resources {
+			if res.ID == nil {
+				continue
+			}
+			resourceId := *res.ID
+			if strings.Contains(resourceId, "providers/Microsoft.App/containerApps") {
```

Comment:
I'm wondering if this ends up matching a nested resource of `providers/Microsoft.App/containerApps/<name>/<child resource>`?

Using `arm.ParseResourceID` could help here. From the resulting `resId`, you could check `resId.ResourceType.Type ==  Microsoft.App/containerApps &&  resId.ResourceType.Types == nil`.

Reply:
It's fine to start here with a simple validation.  Most likely we will not need more than that.